test/rubygems/test_gem_specification.rb

```diff
@@ -3251,7 +3251,7 @@ def test_metadata_validates_ok
     Dir.chdir @tempdir do
       @m1 = quick_gem 'm', '1' do |s|
         s.files = %w[lib/code.rb]
-        s.metadata = { "one" => "two", "home" => "https://example.com/user/repo"  }
+        s.metadata = { "one" => "two", "homepage_uri" => "https://example.com/user/repo"  }
```

Comment:
I understand the need for test, as in if correct keys were not used links won't show up on rubygems.org.
It's hard to imagine such test. `un-prefixed keys` won't be invalid as `metadata` accepts any value for its keys.